pyartcd/pyartcd/pipelines/scheduled/schedule_ocp4_scan.py

```diff
@@ -0,0 +1,40 @@
+import asyncio
+
+import click
+
+from pyartcd import util, jenkins
+from pyartcd.cli import cli, click_coroutine, pass_runtime
+from pyartcd.locks import Lock, LockManager
+from pyartcd.runtime import Runtime
+
+
+async def run_for(version: str, runtime: Runtime):
+    # Skip if locked
+    lock = Lock.BUILD
```

Comment:
I see we're locking on `"github-activity-lock-${cv}"` in https://github.com/openshift-eng/aos-cd-jobs/pull/4226/files , so this should be changed? If i understand the github-activity-lock is the least that's needed for scan-sources / rebase. Also ocp4_scan schedules rhcos build which is not dependent on build lock. So I think we should not ocp4_scan activities like priv rebase / rhcos builds on build lock

Reply:
IIRC, back when we had ocp4 in Groovy/aos-cd-jobs `github-activity-lock-{version}` was shared among ocp4 and ocp4-scan (and others). When ocp4 was moved to pyartcd, and with the adoption of Redis locks, we renamed its locks to `lock:build:{version}`. ocp4-scan is currently the only place where we still use the `github-activity-lock`s.

I agree that having RHCOS builds live under a distinct lock would make sense. Ideally, we might setup a `rhcos-scan` job that only deals with rhcos, and set it to share a common `lock:build:rhcos` lock with the `build/rhcos` pipeline.

Noting this as a follow-up, for the moment I'll just introduce a new `SCAN` Redis lock

Reply:
Hmm you're right. This might lead to some weird things, if I remember ocp4 job can also trigger scan sources which then can also rebase? If ocp4 job only relies on "lock:build" and ocp4_scan relies on "lock:scan" then we have a conflict

Reply:
In my mind these are the distinct write operations we are doing that shouldn't conflict: scan-sources rebase, triggering rhcos build, ocp4 rebase&build

Reply:
> ocp4 job can also trigger scan-sources

`ocp4` does not trigger a ocp4-scan, it only triggers rhcos and sync-for-ci

Reply:
> In my mind these are the distinct write operations we are doing that shouldn't conflict: scan-sources rebase, triggering rhcos build, ocp4 rebase&build

scan-sources rebase deals with rebasing into -priv org, which has nothing to do with doozer rebase. The former only acts on GitHub, the latter pushes to distgit.

By changing the ocp4-scan lock to a new `SCAN` type, I simply purged the `github-activity-lock` which is no longer descriptive, and moved it to pyartcd with a new name. So far, no changes in locking behavior. Once we merge this, we can revisit the way we trigger rhcos (for example, by having a dedicated scan job) and see if we can optimize the locking logics

Reply:
This is what I meant by ocp4 triggering scan-sources" https://github.com/openshift-eng/art-tools/blob/cb30548ded4d1dbf7f8711e203ae4bbbf0ce1db2/pyartcd/pyartcd/pipelines/ocp4.py#L946

Reply:
but maybe i'm mistaken and ocp4 cannot do rebase (controlled by flag?)?